File: C/calculator.c
#include <stdio.h>

float calc(float num1, float num2, char op) {
    float res = 0;
    switch (op)
    {
    case '-': res = num1 - num2;
        break;
    case '+': res = num1 + num2;
        break;
    case '*': res = num1 * num2;
        break;
    case '/': res = num1 / num2;
        break;
    default: printf("%s", "Operation error ");
        break;
    }
    return res;
}


int main(void) {
    float num1;
    float num2;
    char op;
    printf("%s", "Write a: ");
    scanf("%f", &num1);
    printf("%s", "Write b: ");
    scanf("%f", &num2);
    printf("%s", "Write op (+-/*): ");
    scanf(" %c", &op);

    if (op == '/' && num2 == 0) {
        printf("%s", "ZeroDivisionError");
        return 0;
    }

    double answer = calc(num1, num2, op);
    printf("Answer: %f", answer);
    return 0;
}

File: Java/Calculator.java
import java.util.Scanner;

class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Write a: ");
        float num1 = scanner.nextFloat();
        System.out.print("Write b: ");
        float num2 = scanner.nextFloat();
        System.out.print("Write op: ");
        char op = scanner.next().charAt(0);
        float answer = calc(num1, num2, op);
        System.out.printf("Answer: %f", answer);
        scanner.close(); 
    }
    public static float calc(float num1, float num2, char op) {
        float res = 0;
        switch (op) {
            case '-': res = num1 - num2; break;
            case '+': res = num1 + num2; break;
            case '*': res = num1 * num2; break;
            case '/': res = num1 / num2; break;
            default: System.out.println("Operation Error");
                break;
        }
        return res;
    }
}